twenty-first/src/util_types/mmr/mmr_membership_proof.rs

```diff
@@ -306,22 +339,19 @@ impl<H: AlgebraicHasher> MmrMembershipProof<H> {
         modified
     }
 
-    /// Update a membership proof with a `leaf_mutation` proof. For the `membership_proof`
-    /// parameter, it doesn't matter if you use the old or new membership proof associated
-    /// with the leaf update, as they are the same before and after the leaf mutation.
+    /// Update a membership proof with a `leaf_mutation` proof.
```

Comment:
Describing a `LeafMutation` as a proof is misleading. At best, it contains a proof -- and the new leaf and index that is also part of this structure, are certainly necessary for the current update.

Reply:
Agreed.